core/src/main/scala/kafka/server/KafkaApis.scala

```diff
@@ -900,6 +899,57 @@ class KafkaApis(val requestChannel: RequestChannel,
     }
   }
 
+  def handleCreateTopicRequest(request: RequestChannel.Request) {
+    val createTopicRequest = request.body.asInstanceOf[CreateTopicRequest]
+
+    def handleUnauthorizedRequest = {
```

Comment:
We can do that, but I like handling it explicitly. I think this is logic the handlers should own, not the request/response objects. Already, many handlers have to take care of it explicitly and don't even use their getErrorResponse() functions. This is because there are multiple error codes.

Fore example when I add delete topic. We need to send a response code back for each topic. If I have 3 topics; one may have succeeded, another may not exist, and another may be unauthorized. For this reason one exception can't cover all of them, and getErrorResponse() can't have the context needed to build the correct response. Therefore, it must and should be done manually.

This is true of many other responses. To be uniform, more flexible, and have more lightweight request/response objects, I would like to see this be a part of KafkaApis handlers across the board and not the response object. But that is a separate refactoring.